src/stores/Deals.Async.ts

```diff
@@ -27,31 +27,50 @@ export const fetchMinimumDownPayment = trackLoadingInDealsStore(async () => {
 
 
 export const fetchApproval = trackLoadingInDealsStore(async () => {
-        const result = await financingClient.getApproval(
-                dealsStore.carModel!,
-                <EnsurancePlanType[]> dealsStore.selectedInsurancePlans.map(plan => plan.type),
-                <number>dealsStore.downpayment
-            );
-
-        // @Ivan Can you please fix this typescript error
-        // const nextMessage = result.message ?? '';
-
-        const nextStatus = {
-            ...defaultDealStatus,
-            expirationTimer: result.isApproved ? 15 : 0,
-            isApproved: result.isApproved,
-            messages: ['approval has been requested']
-        };
-        
-        dealsStore.setStatus(nextStatus);
-
-        setInterval(() => {
-            if (dealsStore.status?.expirationTimer && dealsStore.status.expirationTimer > 0) {
-                const timeLeft = dealsStore.status.expirationTimer - 1;
-                dealsStore.setStatus({
-                    ...nextStatus,
-                    expirationTimer: timeLeft
-                });
-            }
-        }, 1000)
+    const result = await financingClient.getApproval(
+        dealsStore.carModel!,
+        <EnsurancePlanType[]>dealsStore.selectedInsurancePlans.map(plan => plan.type),
+        <number>dealsStore.downpayment
+    );
+
+    // @Paul, yeah this is something TS team could improve
+    // tagged types don't yet play well with conditionals
+    const nextMessage = result.isApproved === false ? [result.message] : [];
+
+    const nextStatus = {
+        ...defaultDealStatus,
+
+        // this is very bad. What if request took 10 seconds to get back to us from the serve?
+        // Sending bare dates is can also be problematic in real world due to client time being off by a few minutes
+        // but for the sake of our excercies, lets assume client and server clock are both correct 
```

Comment:
I agree, I have already implemented that correctly in version 3 and will fix this here too when version-3 is finished.